extensions/amp-story/1.0/amp-story.js

```diff
@@ -1608,10 +1610,20 @@ export class AmpStory extends AMP.BaseElement {
         'amp-audio, amp-video:not([noaudio]), [background-audio]');
     const hasStoryAudio = this.element.hasAttribute('background-audio');
 
-    this.storeService_.dispatch(Action.TOGGLE_HAS_AUDIO,
+    this.storeService_.dispatch(Action.TOGGLE_STORY_HAS_AUDIO,
         containsMediaElementWithAudio || hasStoryAudio);
   }
 
+  /**
+   *
+   * @private
+   */
+  updateSoundPage_() {
+    const hasPageAudio = this.activePage_.element.hasAttribute('background-audio');
```

Comment:
There are actually several cases where the page has audio:

- `amp-story-page` has a `background-audio` attribute
- `amp-story-page` has a at least one video that doesn't have a `noaudio` attribute
- not sure about this one, but I think a publisher could also use a `amp-audio` component in a page

Maybe this code should live in the `amp-story-page.js` file, that provides some methods like `getAllVideos_` or `getAllMedia_` so it is easier for you to check if the page has sound.

I think the best place for you to check if a page has audio, and maybe dispatch the action, would be in the `if (is page active)` of the `resumeCallback` of `amp-story-page.js`: https://github.com/ampproject/amphtml/blob/master/extensions/amp-story/1.0/amp-story-page.js#L281